### PR TITLE
HomeTask3

### PR DESCRIPTION
**Pull request by BayMaxMB**

### CODE REVIEW

File: app2.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+require('dotenv').config();
```
Comment (Lokki17) [inline app2.js]: what does mean file name 'app2.js'?

File: app2.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+app.use(express.urlencoded({ extended: false }));
+
+app.get('/users', async(req, res) => {
+	const users = await User.find({});
+	res.send(users);
+});
+
+app.post('/users', async(req, res) => {
+	try {
+		const user = await User.create(req.body);
```
Comment (Lokki17) [inline app2.js]: Dont flood routes by business logic. Add controllers to handle requests and services for business logic

File: models/accounts.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	},
+	currency: {
+		type: String,
+		required: true
+	},
+	ammount: {
+		type: Number,
+		default: 0
+	},
+	income: {
```
Comment (Lokki17) [inline models/accounts.js:17]: It's very bad idea to store incomes and expences in account. Move to another entity

File: models/accounts.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+userSchema.methods.edit = function(title, currency) {
+	console.log(`Hi, I'm ${this.name}`);
+}
+
+userSchema.statics.register = async function(user) {
+	const temp = {
+		email: user.email,
+		password: bcrypt.hashSync(user.password, 10),
+		role: user.role,
+		credentials: {
```
Comment (Lokki17) [inline models/accounts.js:40]: Why do you need field 'credentials'? Why dont store first_name, last_name and so on in root?

File: models/users.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+		gender: {
+			type: String,
+			required: true
+		},
+		country_of_residence: {
+			type: String,
+			required: true
+		},
+	},
+	accounts: {
```
Comment (Lokki17) [inline models/users.js:39]: Remove this field. Account should have link to user

File: models/users.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+				default: []
+			},
+			expence: {
+				type: [{}],
+				default: []
+			},
+		}],
+		default: []
+	},
+	categories: {
```
Comment (Lokki17) [inline models/users.js:54]: The same. Delete

File: models/users.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+	// },
+	// edited: {
+	// 	type: Number,
+	// 	default: 0
+	// }
+}, {
+	timestamps: true
+});
+
+userSchema.methods.sayHi = function() {
```
Comment (Lokki17) [inline models/users.js]: Why do you need this method?

File: models/users.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+userSchema.statics.login = async function(email, password) {
+	const user = await this.findOne({ email });
+	console.log(user)
+	if (user && bcrypt.compareSync(password, user.password)) {
+		return user;
+	}
+	return null;
+}
+
+userSchema.statics.findByEmail = async function(email) {
```
Comment (Lokki17) [inline models/users.js]: move methods (business logic) to service

File: routes/accounts.js [hunk 9]
```diff
@@ -1,17 +1,26 @@
 const express = require('express');
 const router = express.Router();
 const { auth } = require('../guards');
+const User = require('../models/users');
 
-router.get('/', auth, (req, res) => {
-	res.json(req.user.accounts);
+router.get('/', auth, async(req, res) => {
+	const user = await User.findOne({ email: req.user.email });
```
Comment (Lokki17) [inline routes/accounts.js]: Dont flood routes. Add Controllers and Services

**Review by Lokki17 — CHANGES_REQUESTED** [review]
Clean up routes, fix remarks

**Comment by BayMaxMB** [comment]
Moving bussiness logic from schema file into services file: means likning functions from services file into schema file?

What about controllers? is that routes files? or it is separate file? what are the jobs of controller?

**Comment by Lokki17** [comment]
> Moving bussiness logic from schema file into services file: means likning functions from services file into schema file?
> 
> What about controllers? is that routes files? or it is separate file? what are the jobs of controller?

Controllers has functions which handle requests: get request params and pass them to service, wait for results and then send results

**Pull request merged**